src/librustc_query_system/query/job.rs

```diff
@@ -452,7 +452,9 @@ fn remove_cycle<CTX: QueryContext>(
 
         // Remove the queries in our cycle from the list of jobs to look at
         for r in &stack {
-            jobs.remove_item(&r.1);
+            if let Some(pos) = jobs.iter().position(|j| j == &r.1) {
+                jobs.remove(pos);
+            }
```

Comment:
@Zoxc git says you were most recently involved with this piece of code and I have a question ^_^ 

Currently my change just replaces `remove_item` with the exact thing that `remove_item` did before. However, I wonder if a `swap_remove` would be fine here, too. Is the order of elements in `jobs` relevant? `swap_remove` could make this a lot faster. 

Furthermore, this O(|stack| * |jobs|) runtime bothers me. I wonder if that can be improved. Is one of the two vectors generally shorter than the other one?

Reply:
IIRC, cycles should generally not occur so this code need not be optimized. I wouldn't worry about it. In any case it's parallel-only so it doesn't matter too much at present (we don't ship that configuration to anyone).